### PR TITLE
Grant acces to dataset

### PR DESCRIPTION
**Pull request by robertmunne**

### CODE REVIEW

File: src/test/resources/features/GrantDatasetAccess.feature [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    And I create a new dataset with title "title1" and description "description1"
+    And There is a registered reuser with username "reuser" and password "passwd" and email "reus@gmail.com"
+    And I login as "reuser" with password "passwd"
+    And I create a new request with description "description1"
+    And There is a registered provider with username "provider" and password "password" and email "prov@gmail.com"
+    And I login as "provider" with password "password"
+    And I create a new DatasetRequest associate with Dataset "title1" and "description1" and Request "description1" with status value "True"
+    And It has been created a new DatasetRequest
+    When I change the status value of DatasetRequest associate with Dataset "title1" and "description1" and "True" to "False"
+    Then The response code is 200
```
Comment (bergacat1) [inline src/test/resources/features/GrantDatasetAccess.feature:19]: Eventually, you should check that the status has been changed correctly.

File: src/test/resources/features/GrantDatasetAccess.feature [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+
+  Scenario: Provider tries to change DatasetRequest
+    Given There is a created DatasetRequest associate with Dataset "title1" and "description1" and Request "description1" with status value "True"
+    And There is a registered provider with username "provider" and password "password" and email "prov@gmail.com"
+    And I login as "provider2" with password "password"
+    When I change the status value of DatasetRequest associate with Dataset "title1" and "description1" and "True" to "False"
+    Then The response code is 401
```
Comment (bergacat1) [inline src/test/resources/features/GrantDatasetAccess.feature:27]: The Provider with username "provider2" is not in the database. That's why it returns a response code of 401.

**Review by bergacat1 — DISMISSED** [review]
Additionally, the status of the DatasetRequest should be 'false' by default and Providers should change it to 'true' to grant access to Reusers.

**Comment by rogargon** [comment]
Still missing the most important scenario, to check that other users beyond the original provider cannot grant access to a dataset provided by the provider. Check with, in addition to the anonymous user, a reuser or another provider

File: src/test/resources/features/GrantDatasetAccess.feature [hunk 3]
```diff
@@ -28,5 +28,22 @@ Feature: Grant Dataset Access
...
+    When I change the status value of DatasetRequest associate with Dataset "title1" and "description1" and "False" to "True"
+    Then The response code is 401
+
+  Scenario: provider tries to change DatasetRequest value
+    Given There is a registered provider with username "provider" and password "password" and email "prov1@gmail.com"
+    And There is a created dataset with title "title1" and description "description1"
+    And There is a created DatasetRequest associate with Dataset "title1" and "description1" and Request "description1" with status value "False"
+    And I login as "provider1" with password "password1"
+    When I change the status value of DatasetRequest associate with Dataset "title1" and "description1" and "False" to "True"
+    Then The response code is 401
```
Comment (bergacat1) [inline src/test/resources/features/GrantDatasetAccess.feature:45]: You have not registered the Provider with username "provider1". For this reason, the application is returning a 401.

**Review by rogargon — APPROVED** [review]

**Pull request merged**